### ISSUE TITLE
php-cs-fixer > 3.9.5 causes LogicException in caching layer

### ISSUE THREAD

**Issue by billypoke** [nunomaduro/phpinsights#593]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| Library version  | 2.4.0

Pinning `friendsofphp/php-cs-fixer` at `3.9.5` works as a temporary solution. `3.9.6` and `3.10.0` are both giving the error

https://github.com/nunomaduro/phpinsights/blob/master/src/Domain/Runner.php#L183 This line is the source of the error

_Originally posted by @billypoke in https://github.com/nunomaduro/phpinsights/issues/511#issuecomment-1219588656_

**Comment by real-splendid** [comment]
Recently, in phpinsights dependency (FriendsOfPHP/PHP-CS-Fixer) they substituted php-cs-fixer/diff with sebastian/diff. Since phpinsights used php-cs-fixer/diff and it was not locked in composer.json, It does not work anymore. This problem appeared as "LogicException Unable to find data for ...".

already fixed it in my PR
https://github.com/nunomaduro/phpinsights/pull/591

**Comment by real-splendid** [comment]
> https://github.com/nunomaduro/phpinsights/blob/master/src/Domain/Runner.php#L183 This line is the source of the error
> 
> _Originally posted by @billypoke in [#511 (comment)](https://github.com/nunomaduro/phpinsights/issues/511#issuecomment-1219588656)_

https://github.com/nunomaduro/phpinsights/blob/984c1b333afcda5d2586d4b4a389939b80862900/src/Domain/Differ.php#L7 this line is the source of the error

and [#511 (comment)](https://github.com/nunomaduro/phpinsights/issues/511#issuecomment-1219588656) probably have a different source of error. 
LogicException may occur when `Runner` start `Symfony\Component\Process\Process` and it may fail for different reasons.
So it's good idea to have check `$process->getExitCode()` somewhere and throw exception with text from  `$process->getErrorOutput()`

**Comment by patrickomeara** [comment]
I'm also having this issue. Thanks for writing it up @billypoke 

To pin 3.9.5: ```composer require friendsofphp/php-cs-fixer:v3.9.5```

**Comment by FulgerX2007** [comment]
> I'm also having this issue. Thanks for writing it up @billypoke
> 
> Strangely enough it's only when I run insights with `--fix` or `--disable-security-check`
> 
> To pin 3.9.5: `composer require friendsofphp/php-cs-fixer:v3.9.5`

Thank you, it helps.

**Comment by renky** [comment]
> 

I also had this problem with a nearly complete clean setup of a new laravel-project. I got the message 
```
LogicException 

Unable to find data for /var/www/routes/console.php

at vendor/nunomaduro/phpinsights/src/Domain/Runner.php:184
```

Pinning php-cs-fixer to version 3.9.5 helped.

**Comment by cmgmyr** [comment]
This should be fixed in `2.5.0`. Please let us know if you are still having issues

**Comment by realtek2** [comment]
@cmgmyr 
Php Insights version 2.8.0 the same error.
